tests/e2e/multi_vc_preferential_topology.go

```diff
@@ -0,0 +1,776 @@
+	    10. Clear the data
+	*/
+	ginkgo.It("Tag one datastore as preferred each in VC1 and VC2 and verify it is honored", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		parallelStatefulSetCreation = true
+		preferredDatastoreChosen = 1
+		preferredDatastorePaths = nil
+		sts_count := 3
+		stsReplicas = 10
+		var dsUrls []string
+		scaleUpReplicaCount = 15
+		stsScaleDown = false
+
+		ginkgo.By("Create SC with default parameters")
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, nil, nil, "",
+			"", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// choose preferred datastore
+		ginkgo.By("Tag preferred datastore for volume provisioning in VC1 and VC2")
+		for i := 0; i < 2; i++ {
+			paths, err := tagPreferredDatastore(masterIp, sshClientConfig, allowedTopologies[0].Values[i],
+				preferredDatastoreChosen, ClusterdatastoreListVC[i], nil, true, i)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			preferredDatastorePaths = append(preferredDatastorePaths, paths...)
+
+			// Get the length of the paths for the current iteration
+			pathsLen := len(paths)
+
+			for j := 0; j < pathsLen; j++ {
+				// Calculate the index for ClusterdatastoreListVC based on the current iteration
+				index := i + j
+
+				if val, ok := ClusterdatastoreListVC[index][paths[j]]; ok {
+					dsUrls = append(dsUrls, val)
+				}
+			}
+		}
+
+		framework.Logf("Waiting for %v for preferred datastore to get refreshed in the environment",
+			preferredDatastoreTimeOutInterval)
+		time.Sleep(preferredDatastoreTimeOutInterval)
+
+		ginkgo.By("Create service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+
+		ginkgo.By("Create 3 statefulset with 10 replicas")
+		statefulSets := createParallelStatefulSetSpec(namespace, sts_count, stsReplicas)
+		var wg sync.WaitGroup
+		wg.Add(sts_count)
+		for i := 0; i < len(statefulSets); i++ {
+			go createParallelStatefulSets(client, namespace, statefulSets[i],
+				stsReplicas, &wg)
+
+		}
+		wg.Wait()
+
+		ginkgo.By("Waiting for StatefulSets Pods to be in Ready State")
+		err = waitForStsPodsToBeInReadyRunningState(ctx, client, namespace, statefulSets)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			deleteAllStsAndPodsPVCsInNamespace(ctx, client, namespace)
+		}()
+
+		ginkgo.By("Verify PV node affinity and that the PODS are running on appropriate node")
+		for i := 0; i < len(statefulSets); i++ {
+			verifyPVnodeAffinityAndPODnodedetailsForStatefulsetsLevel5(ctx, client, statefulSets[i],
+				namespace, allowedTopologies, parallelStatefulSetCreation, true)
```

Comment:
Can't we do this in a single for loop ie.  verifyPVnodeAffinityAndPODnodedetailsForStatefulsetsLevel5 and verifyVolumeProvisioningForStatefulSet

Reply:
But here we are creating 3 StatefulSets and both the util takes 1 StatefulSet at a time. Hence, used for loop here